hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -777,7 +778,7 @@ static HoodieData<HoodieRecord> convertMetadataToRecordIndexRecords(HoodieEngine
     List<HoodieWriteStat> allWriteStats = commitMetadata.getPartitionToWriteStats().values().stream()
         .flatMap(Collection::stream).collect(Collectors.toList());
 
-    if (allWriteStats.isEmpty()) {
+    if (allWriteStats.isEmpty() || commitMetadata.getOperationType() == WriteOperationType.COMPACT) {
```

Comment:
we were not updating RLI during compaction before: 
https://github.com/apache/hudi/pull/12321/files#r1858051128
So, following the same.

Reply:
yes, I have added a test to ensure RLI does not generate any records for compaction commits.